File: Exemplos/ExemploPolimorfismo.java
public class ExemploPolimorfismo {
    
}

public class Pessoa {

  String nome, time;
  
  public Pessoa() {

    System.out.println("Pessoa criada");
        
  }
  
  public Pessoa(String nome, String time) {
      
    this.nome = nome;
    this.time = time;
    System.out.println("Pessoa criada");
      
  }
  
  public void apresentacao () {
      
    System.out.println(this.nome + " é uma pessoa que gosta do time " + this.time + ".");
    
  }
  
}

public class Jogador extends Pessoa {
  
  String apelido, posicao;
  float salario;
  int camisa;
    
  public Jogador() {

    System.out.println("Jogador criado");
        
  }
    
  public Jogador(String nome, String time, String apelido, String posicao, int camisa, float salario) {
  
    super (nome, time);
    this.apelido = apelido;
    this.posicao = posicao;
    this.camisa = camisa;
    this.salario = salario;
    System.out.println("Jogador criado");
        
  }
  
  @Override
  public void apresentacao () {
      
    System.out.println(this.nome + " é o jogador camisa " + this.camisa + " do time " + this.time + ".");
    
  }
    
}

public class Tecnico extends Pessoa {

  float salario;
  
  public Tecnico() {

    System.out.println("Técnico criado");
        
  }
  
  public Tecnico(String nome, String time, float salario) {
  
    super (nome, time);
    this.salario = salario;
    System.out.println("Técnico criado");
      
  }
  
  @Override
  public void apresentacao () {
      
    System.out.println(this.nome + " é o técnico do time " + this.time + ".");
    
  }
  
}

public class Torcedor extends Pessoa {
  
  public Torcedor() {

    System.out.println("Torcedor criado");
      
  }

  public Torcedor(String nome, String time) {

    super (nome, time);
    System.out.println("Torcedor criado");
    
  }
  
  @Override
  public void apresentacao () {
      
    System.out.println(this.nome + " é um torcedor do time " + this.time + ".");
    
  }

}

File: Exemplos/ExemploPolimorfismo.py
class Pessoa:

    def __init__(self, nome, time):
        self._nome = nome
        self._time = time

    def apresentacao(self):
        print(self._nome + " é uma pessoa que gosta do time " + self._time ".")


class Técnico(Pessoa):

    def __init__(self, nome, time, salario):
        super().__init__(nome, time)
        self.__salario = salario
        print("Técnico Criado")

    def apresentacao(self):
        print(self._nome + " é o técnico do time " + self._time ".")


class Jogador(Pessoa):

    def __init__(self, nome, time, apelido, posicao, camisa, salario):
        super().__init__(nome, time)
        self.__apelido = apelido
        self.__posicao = posicao
        self.__camisa = camisa
        self.__salario = salario
        print("Jogador Criado")

    def apresentacao(self):
        print(self._nome + " é o jogador camisa" + self.__camisa + " do time " + self._time ".")

class Torcedor(Pessoa):

    def __init__(self, nome, time):
        super().__init__(nome, time)
        print("Torcedor Criado")

    def apresentacao(self):
        print(self._nome + " é um torcedor do time " + self._time ".");
